.github/workflows/terragrunt-apply-production.yml

```diff
@@ -145,31 +178,49 @@ jobs:
         working-directory: env/cloud/app
         run: terragrunt apply --terragrunt-non-interactive -auto-approve
 
-      - name: Install Lambda deps
-        run: ./aws/lambdas/code/deps.sh install
-
       - name: Terragrunt apply lambdas
         working-directory: env/cloud/lambdas
         run: terragrunt apply --terragrunt-non-interactive -auto-approve
 
-      - name: Remove Lambda deps
-        run: ./aws/lambdas/code/deps.sh delete
-
-      - name: Install Lambda Alarm deps
-        run: ./aws/alarms/lambda/deps.sh install
-
       - name: Terragrunt apply alarms
         working-directory: env/cloud/alarms
         run: terragrunt apply --terragrunt-non-interactive -auto-approve
 
-      - name: Remove Lambda deps
-        run: ./aws/alarms/lambda/deps.sh delete
+  update-lambda-function-image:
+    needs: [detect-lambda-changes, build-tag-push-lambda-images, terragrunt-plan]
+    if: needs.detect-lambda-changes.outputs.lambda-to-rebuild != '[]'
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        image: ${{ fromJSON(needs.detect-lambda-changes.outputs.lambda-to-rebuild) }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
 
-      - name: Notify Slack on failure
-        if: failure()
+      - name: Request Lambda functions to use latest image
+        uses: ./.github/workflows/request-lambda-functions-to-use-latest-image
+        with:
+          aws-role-to-assume: arn:aws:iam::${{ env.AWS_ACCOUNT_ID }}:role/forms-terraform-apply-release
+          aws-role-session-name: TFApply
+          aws-region: ${{ env.AWS_REGION }}
+          lambda-name: ${{ matrix.image }}
```

Comment:
One thing that could happen here if you're changing a function's config is that between the time that Terraform updates the function in the `terragrunt-apply-all-modules` job and this deployment, you may have errors as the old function Docker image tries to handle the new function config.

Reply:
True! Did not think about this specific case.
There would a problem if we remove an environment variable which the opposite of the scenario (add a new env var) I tried to handle in this deployment logic.
What we could do to prevent this from happening is to make sure we do not delete unused env var right away. We could let them in until we have released in production.